src/System.IO.FileSystem/tests/FileSystemTest.cs

```diff
@@ -21,11 +21,22 @@ protected static bool CanCreateSymbolicLinks
         {
             get
             {
+                // Verify file symlink creation
                 var path = Path.GetTempFileName();
                 var linkPath = path + ".link";
                 var ret = MountHelper.CreateSymbolicLink(linkPath, path, isDirectory: false);
+                ret = ret && File.Exists(linkPath);
```

Comment:
>In what situation will CreateSymbolicLink return true but the link won't have been created?

none. If the link isn't created then teh return should be false. We could quit whenever `ret` first hits a `false` which would occur if the CreateSymbolicLink helper failed. I'll refactor it to exit whenever it his a failed operation.